### PR TITLE
Fix potential problems in ResourceResolver.loadJSONFromAsset()

### PR DESCRIPTION
**Pull request by sumio**
## Issue

`ResourceResolver.loadJSONFromAsset()` has the following code:

```java
            is = context.getAssets().open("json/" + jsonFileName);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            json = new String(buffer, "UTF-8");
```

The above code may read only partial content of the file because `is.available()` may return less than the length of the file.

## Overview (Required)

- Use Apache Commons IO to avoid handling such a complicated stream behavior.

## Links

N/A

## Screenshot

N/A

### CODE REVIEW

**Comment by eyedol** [comment]
@sumio good call on the use of `available()`. However, I think it's much better to use `Okio` to handle file reads as it's already a dependency. Something like the code below should work. I haven't read `Okio` source code yet but I believe it solves that issue. If not feel free to ignore this.

```java
String json = null;
BufferedSource source = null;
try {
    final InputStream is = context.getAssets().open("json/" + jsonFileName);
    source = Okio.buffer(Okio.source(is));
    json = source.readUtf8();
} catch (IOException e) {
    Timber.tag(TAG).e(e, "assets/json/%s: read failed", jsonFileName);
} finally {
    try {
        if (source != null) {
            source.close();
        }
    } catch (IOException e) {
        Timber.tag(TAG).e(e, "assets/json/%s: read failed", jsonFileName);
    }
}
return json;       
```

**Comment by konifar** [comment]
:eyes:

**Comment by sumio** [comment]
@eyedol Thanks for your nice suggestion!
I'll rewrite it by using Okio in this night.

**Comment by sumio** [comment]
Rewriting is done!
commons-io dependecy is removed.

**Comment by sumio** [comment]
It seems that circleCI is failed in `ContributorsLocalDataSourceTest > updateAllAsyncAsInsert` which is unrelated in this PR...
(and all tests are passed in my local environment)

**Comment by konifar** [comment]
@sumio Sorry, CircleCI is not stable now 🙇

**Comment by sumio** [comment]
@konifar Thanks for your reply.
OK. I won't mind the result.

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java [hunk 1]
```diff
@@ -33,22 +35,20 @@ public String getString(@StringRes int resId, Object... formatArgs) {
...
-        InputStream is = null;
+        BufferedSource bufferedSource = null;
         try {
-            is = context.getAssets().open("json/" + jsonFileName);
-            int size = is.available();
-            byte[] buffer = new byte[size];
-            is.read(buffer);
-            json = new String(buffer, "UTF-8");
+            InputStream is = context.getAssets().open("json/" + jsonFileName);
+            bufferedSource = Okio.buffer(Okio.source(is));
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java]: 🆒

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java [hunk 2]
```diff
@@ -33,22 +35,20 @@ public String getString(@StringRes int resId, Object... formatArgs) {
...
         } finally {
             try {
-                if (is != null) {
-                    is.close();
+                if (bufferedSource != null) {
+                    bufferedSource.close();
                 }
             } catch (IOException e) {
-                Timber.e(TAG, e.getMessage(), e);
+                // ignore
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java]: 🙋 Why did you ignore here?

**Comment by konifar** [comment]
@sumio Sorry for late review  🙇 LGTM! Thanks for contribution! 💯 
@eyedol Thanks for great suggestion! I appreciate you!

**Comment by konifar** [comment]
Hope CI success 🙏

**Comment by konifar** [comment]
👏

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java (hunk 2 above)
Comment (sumio) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java]: Personally I think, it is commonly used that failure in finally block is ignored quietly.
(cf. [IOUtils.closeQuietly](https://commons.apache.org/proper/commons-io/javadocs/api-release/org/apache/commons/io/IOUtils.html#closeQuietly%28java.io.Closeable%29))

That said, it should be respected that project owners' opinion. 
Shall I put the log?

Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java]: Understood! 📝 I follow your opinion. Thanks for IOUtils example!

**Pull request merged**